Psi4 error reporting pre-empted by KeyError

dotsdl opened the pull request:
<!-- Thank you for your contribution! -->

## Description
<!-- Provide a brief description of the PR's purpose here. -->

Currently, useful error reporting with `psi4` under the `Psi4Harness` can be lost due to a missing `"success"` key on `output_data`. This PR makes attempted access of this key return `False` if not present.

Credit to @dgasmith for pointing this out.

## Changelog description
<!-- Provide a brief single sentence for the changelog. -->
Error messages from `Psi4Harness` no longer swallowed by `KeyError`.

## Status
<!-- Please `pip install .[lint]; make format` in the base folder. -->
- [x] Code base linted
- [x] Ready to go

dgasmith commented on qcengine/programs/psi4.py:
```diff
@@ -164,7 +164,7 @@ def compute(self, input_model: "AtomicInput", config: "TaskConfig") -> "AtomicRe
                             output_data["extras"]["qcvars"] = local_qcvars
 
-                    if output_data["success"] is False:
+                    if output_data.get("success", False) is False:
```
To be safe you may want to protect the below dict getters in a similar fashion `error_message = output_data.get("error", "Unknown error, error message is not found")` or whatnot.

dotsdl replied:
Ah thanks! Will add this now.

dotsdl commented:
Looks like we're also being hit by the latest black update here.

loriab commented:
The CI failure is for the new code, right?

On the `black` changes, I think let's try to merge some PRs, then do a separate one for formatting.

dotsdl commented:
@loriab ah yes, you're right. It looks like my changes to the error message pathways mangled what is expected in those tests. I'll dive into these to find what's going awry.

dotsdl commented:
Found the issue that was causing tests to fail; was an inconsistency in the error handling logic. Fixed and moved to its own function so we don't have the same delicate decision tree in two places. 

Should be good to go if all tests pass on Travis!

dotsdl commented:
@loriab this one is ready to go! Should improve our error reporting from psi4 failures substantially!

dgasmith reviewed (approved):
Nice looks pretty bullet proof. It's amazing what you end up with to catch all of the edge case at QCArchive scale.

loriab commented:
great, thanks, this was the PR I was keen to get in!
